app/views/shared/_improve_this_page.html.erb

```diff
@@ -0,0 +1,41 @@
+<div class="improve-this-page" data-module="improve-this-page">
+  <div class="js-prompt">
+    <h3 class="improve-this-page__is-useful-question">Is this page useful?</h3>
+    <%= link_to contact_govuk_path, { class: 'button js-page-is-useful', role: 'button' } do %>
+      Yes <span class="visually-hidden"> this page is useful</span>
+    <% end %>
+    <%= link_to contact_govuk_path, { class: 'button js-offer-feedback', role: 'button' } do %>
+      No <span class="visually-hidden"> this page is not useful</span>
+    <% end %>
+    <%= link_to contact_govuk_path, { class: 'button js-offer-feedback improve-this-page__anything-wrong', role: 'button' } do %>
+      Is there anything wrong with this page? <span class="visually-hidden"> this page is not useful</span>
+    <% end %>
+  </div>
+  <div class="js-feedback-form js-hidden">
+    <div class="grid-row">
+      <div class="column-two-thirds">
+        <div class="js-errors"></div>
+        <form>
+          <input type="hidden" name="path" value="<%= request.original_url -%>">
+          <input type="hidden" name="user_agent" value="<%= request.user_agent -%>">
```

Comment:
Yeah, they should do, unless Rails has made those strings "safe", which I doubt it has. I think it's OK, even desirable, having those values HTML escaped isn't it? If someone has put a script tag in their user agent I quite fancy escaping it.. Ampersands will be escaped in a query string in the URL but we don't have paths based on query strings.. I'm not feeling a lot of urgency to make a change here. Any feelings?